app/models/offer.rb

```diff
@@ -54,13 +81,35 @@ def withdraw
   end
 
   def cancel
-    self.status = :cancelled
-    save
+    case status
+    when 'cancelled'
+      # Already cancelled
+      true
```

Comment:
This will work and is clean. Possibly returning nil would be more descriptive given that no action was taken, and could be relayed in UX, or would indicate to admin that no changes were made if command sent via console.

Reply:
Thanks for the thoughts.
These methods all return either true (success) or false (not success), so I'm inclined to leave as-is for this iteration.